test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,1675 @@
+            self.du.run_cmd(host, cmd=cmd, stdout=out_fd, sudo=self.sudo)
+
+            if self.anonymize and not skip_anon:
+                self.__anonymize_file(out_path)
+
+        if self.create_tar:
+            self.__add_to_archive(out_path)
+
+    def __convert_flag_to_numeric(self, flag):
+        """
+        Convert a resource's flag attribute to its numeric equivalent
+
+        :param flag: the resource flag to convert
+        :type flag: string
+
+        :returns: numeric value of the resource flag
+        """
+        ATR_DFLAG_USRD = 0x01
+        ATR_DFLAG_USWR = 0x02
+        ATR_DFLAG_OPRD = 0x04
+        ATR_DFLAG_OPWR = 0x08
+        ATR_DFLAG_MGRD = 0x10
+        ATR_DFLAG_MGWR = 0x20
+        ATR_DFLAG_RASSN = 0x4000
+        ATR_DFLAG_ANASSN = 0x8000
+        ATR_DFLAG_FNASSN = 0x10000
+        ATR_DFLAG_CVTSLT = 0x20000
+
+        NO_USER_SET = (ATR_DFLAG_USRD | ATR_DFLAG_OPRD | ATR_DFLAG_MGRD |
+                       ATR_DFLAG_OPWR | ATR_DFLAG_MGWR)
+        READ_WRITE = (ATR_DFLAG_USRD | ATR_DFLAG_OPRD | ATR_DFLAG_MGRD |
+                      ATR_DFLAG_USWR | ATR_DFLAG_OPWR | ATR_DFLAG_MGWR)
+
+        resc_flag = READ_WRITE
+        if "q" in flag:
+            resc_flag |= ATR_DFLAG_RASSN
+        if "f" in flag:
+            resc_flag |= ATR_DFLAG_FNASSN
+        if "n" in flag:
+            resc_flag |= ATR_DFLAG_ANASSN
+        if "h" in flag:
+            resc_flag |= ATR_DFLAG_CVTSLT
+        if "r" in flag:
+            resc_flag &= ~READ_WRITE
+            resc_flag |= NO_USER_SET
+        if "i" in flag:
+            resc_flag &= ~READ_WRITE
+            resc_flag |= (ATR_DFLAG_OPRD | ATR_DFLAG_OPWR |
+                          ATR_DFLAG_MGRD | ATR_DFLAG_MGWR)
+        return resc_flag
+
+    def __convert_type_to_numeric(self, attr_type):
+        """
+        Convert a resource's type attribute to its numeric equivalent
+
+        :param attr_type: the type to convert
+        :type attr_type: string
+
+        :returns: Numeric equivalent of attr_type
+        """
+        PBS_ATTR_TYPE_TO_INT = {
+            "long": 1,
+            "string": 3,
+            "string_array": 4,
+            "size": 5,
+            "boolean": 11,
+            "float": 14,
+        }
+
+        return PBS_ATTR_TYPE_TO_INT[attr_type.strip()]
+
+    def __capture_trace_from_core(self, core_file_name, exec_path, out_path):
+        """
+        Capture stack strace from the core file specified
+
+        :param core_file_name: name of the core file
+        :type core_file_name: str
+        :param exec_path: path to the executable which generated the core
+        :type exec_path: str
+        :param out_path: ofile to print the trace out to
+        :type out_path: str
+        """
+        self.logger.info("capturing stack trace from core file " +
+                         core_file_name)
+
+        # Create a gdb-python script to capture backtrace from core
+        gdb_python = """
+        import gdb
```

Comment:
Thanks for catching this, it looks pretty ugly when not indented, so I now remove the tabs later, let me know what you think.